// @hubspot/ui-extensions/dist/types.js
export class ExtensionEvent {
    type;
    bubbles;
    timeStamp;
    id;
    constructor(event) {
        this.bubbles = event.bubbles;
        this.type = event.type;
        this.timeStamp = event.timeStamp;
        this.id = crypto.randomUUID();
    }
}
export class FormSubmitExtensionEvent extends ExtensionEvent {
    targetValue;
    constructor(value, event) {
        super(event);
        this.targetValue = value;
    }
}
/**
 * @category Serverless
 */
export var ServerlessExecutionStatus;
(function (ServerlessExecutionStatus) {
    ServerlessExecutionStatus["Success"] = "SUCCESS";
    ServerlessExecutionStatus["Error"] = "ERROR";
})(ServerlessExecutionStatus || (ServerlessExecutionStatus = {}));
/** @deprecated use ExtensionEvent/FormSubmitExtensionEvent instead */
export class RemoteEvent {
    type;
    bubbles;
    timeStamp;
    targetValue;
    constructor(value, event) {
        this.bubbles = event.bubbles;
        this.type = event.type;
        this.timeStamp = event.timeStamp;
        this.targetValue = value;
    }
}
